### DISCUSSION TITLE
[Guide] Outpainting in StableSwarmUI 0.6.3

### DISCUSSION THREAD

**Discussion by mcmonkey4eva** [Stability-AI/StableSwarmUI#360] [Announcements]
This page is a quick temporary guide for how to outpaint an image in version 0.6.3 of StableSwarmUI. (In the future more dedicated tooling might be added).

This relies on brand-new features as of the time posted - so if you haven't updated recently, update first - click the "server" tab and click "update and restart server"
![image](https://github.com/Stability-AI/StableSwarmUI/assets/4000772/f4940298-36ca-4e19-afc1-faaeef93ae59)

ps join the [Swarm Discord](https://discord.gg/q2y38cqjNw)

## Step 1: Open The Image Editor

Step 1 is open the image editor -- find your starting image and get it in the center area of Swarm (click it in history, or drag from elsewhere to the center)
Then, click the "Edit Image" button

![image](https://github.com/Stability-AI/StableSwarmUI/assets/4000772/c4d56b35-684e-4488-9301-d7272767936d)

This will open the image editor view next to your image.
Use scroll wheel to zoom out, or click+drag to move around.

![image](https://github.com/Stability-AI/StableSwarmUI/assets/4000772/d2ba6e47-0808-482d-9bdc-1f7bf8ed982b)

## Step 2: Positioning

Next up is repositioning the image.
This step is the particular weird trick: rather than repositioning the 'camera', we need to move everything else to the side. (This will definitely be addressed in the future so you can move the 'camera' (that yellow dashed line)).

First, select the image on the right side layer list
![image](https://github.com/Stability-AI/StableSwarmUI/assets/4000772/7a6bfd6b-77c3-49bc-a004-95d83f9cf796)

Then, select the "Move" tool on the left
![image](https://github.com/Stability-AI/StableSwarmUI/assets/4000772/3ba76b80-8e17-4f75-8230-e8d618404038)

Now with that, hold CTRL (control) and drag the image to the side.
For example here I've decided to outpaint a little bit to the left, and so I need to move the image rightward
![image](https://github.com/Stability-AI/StableSwarmUI/assets/4000772/c48fc042-270e-4f3e-a064-80308e4be342)

## Step 3: Tell Swarm You Want Masking

By default, swarm will regenerate the entire image - we want to regenerate just the empty region, so we need to "tell swarm" that we want the mask (in the future this will probably have a more direct method, but for now) all you need to do is:

select the mask layer
![image](https://github.com/Stability-AI/StableSwarmUI/assets/4000772/fb5f9706-f7ed-4197-9d90-4fd5fc583f68)
select the brush tool
![image](https://github.com/Stability-AI/StableSwarmUI/assets/4000772/c60d3ae1-f6b6-4c6b-9f9a-e6bb8242f7fe)
and draw anything at all off to the side 
![image](https://github.com/Stability-AI/StableSwarmUI/assets/4000772/41a07429-0daf-439d-8770-6f4a8a27e7cc)

While you're at it, feel free to add any real masking you want.
But as long as there's *any* mask, the transparent region (ie the spot that has no image) automatically gets 100% mask.

## Step 4: Configure

On the left, open the "Init Image" param group, set Creativity and Reset To Norm to max,
and set Mask Blur to 0 if you didn't do a manual mask, or raise it if you are doing a manual mask.
![image](https://github.com/Stability-AI/StableSwarmUI/assets/4000772/bc4e0584-90d0-4a2e-b4ff-b85539f7a113)

Also at the bottom type in your prompt
![image](https://github.com/Stability-AI/StableSwarmUI/assets/4000772/b5809f9e-0d76-4319-86f5-48fb7ece6393)

Note you might also want to make sure you're using a model that's decent at inpainting. Any model works, but some yield better results than others. (Remember, outpainting is just inpainting to the side)

## Step 5: Go

Click generate
![image](https://github.com/Stability-AI/StableSwarmUI/assets/4000772/5c60dba2-8bae-48d8-a116-eea4d61a1df4)

Take a look at the result, and if you don't like it change your params/mask/whatever and try again until you're happy.

In my case I decided to go back and change my prompt + draw some partial opacity masks all over to give the AI better working room and try to fix the messed up tail cut off (I'm working with SDXL 1.0 Base, not the best at inpainting)
![image](https://github.com/Stability-AI/StableSwarmUI/assets/4000772/56bbf7a2-3ea0-45ea-8128-10b067d7f09a)

## Step 6: Apply

When you're happy with the result, click+drag the image result onto the canvas
![clickndrag](https://github.com/Stability-AI/StableSwarmUI/assets/4000772/75051240-f660-49a1-91de-a56388f405bb)

Feel free to make any other changes, inpaint things, etc. at will from here

## Step 7: Get the result

Now we have an original image, and an outpaint image, but the only "final result" is the canvas itself - so how do we get the full image?!
(In the future this part will have a more direct method too)

on the left toolbar, click the "\*\*\*" Options menu, and click either "Store Full Canvas To History" or "Download Full Canvas"

![image](https://github.com/Stability-AI/StableSwarmUI/assets/4000772/78668136-bc7a-433f-800f-4bfe787e5c72)

(Note that "Store Full Canvas To History" will immediately pop up the full result image in your right side result area)

and enjoy your final image

![1248-a photo of a black cat in a forest, fluf-OfficialStableDiffusionsd_xl_base_10s-1585428621](https://github.com/Stability-AI/StableSwarmUI/assets/4000772/662a783e-c76c-43d7-84ab-06c607a676c5)

**Comment by CantainPike** [comment]
Thank You very well explained.

**Comment by Arche1987** [comment]
- [ ] []()

**Comment by el-nuru** [comment]
just a black square is generated every time :/

**Reply by Urabewe** [reply]
[SwarmUI](https://github.com/mcmonkeyprojects/SwarmUI)

You're using an old outdated version. This project has moved and been updated significantly. You'll want to go here and use the install bat to install this version. You should be able to move your models and output images to the newer server
